frontend/packages/ceph-storage-plugin/integration-tests-cypress/views/store.ts

```diff
@@ -0,0 +1,69 @@
+export enum Providers {
+  AWS = 'AWS S3',
+  AZURE = 'Azure Blob',
+  S3 = 'S3 Compatible',
+  PVC = 'PVC',
+}
+
+export const testName = 'test-bucket';
+
+export enum StoreType {
+  BackingStore = 'backingstore',
+  NamespaceStore = 'namespacestore',
+}
+
+const inputCustomSecrets = (storeType: StoreType) => {
+  cy.byTestID('switch-to-creds').click();
+  cy.byTestID(`${storeType}-access-key`).type('my_dummy_test_key');
+  cy.byTestID(`${storeType}-secret-key`).type('my_dummy_sec_key');
+  cy.byTestID(`${storeType}-target-bucket`).type('my_dummy_target');
+};
+
+const setupAWS = (storeType: StoreType) => {
+  cy.byTestDropDownMenu(Providers.AWS).click();
+  cy.byTestID(`${storeType}-aws-region-dropdown`).click();
+  cy.byTestDropDownMenu('us-east-1').click();
+  inputCustomSecrets(storeType);
+};
+
+const setupAzureBlob = (storeType: StoreType) => {
+  cy.byTestDropDownMenu(Providers.AZURE).click();
+  inputCustomSecrets(storeType);
+};
+
+const setupS3Endpoint = (storeType: StoreType) => {
+  const ENDPOINT = 'http://test-endpoint.com';
+  cy.byTestDropDownMenu('S3 Compatible').click();
+  cy.byTestID(`${storeType}-s3-endpoint`).type(ENDPOINT);
+  inputCustomSecrets(storeType);
+};
+
+const setupPVC = () => cy.byTestDropDownMenu('PVC').click();
+
+const setupProvider = (provider: Providers, storeType: StoreType) => {
+  cy.byTestID(`${storeType}-provider`).click();
+  switch (provider) {
+    case Providers.AWS:
+      setupAWS(storeType);
+      break;
+    case Providers.AZURE:
+      setupAzureBlob(storeType);
+      break;
+    case Providers.S3:
+      setupS3Endpoint(storeType);
+      break;
+    case Providers.PVC:
+      setupPVC();
+      break;
+    default:
+      break;
+  }
+};
+
+export const STORE = {
+  createStore: (provider: Providers, storeType: StoreType = StoreType.BackingStore) => {
+    cy.byTestID(`${storeType}-name`).type(testName);
+    setupProvider(provider, storeType);
+    cy.byTestID(`${storeType}-create-button`).click();
+  },
```

Comment:
If there is only one key, it better to take out the inside function and avoid object.